Makefile

```diff
@@ -58,32 +56,30 @@ copycomp:
 
 # Build api docs
 updateapispec:
-	cp $(K8SROOT)/api/openapi-spec/swagger.json gen-apidocs/generators/openapi-spec/swagger.json
+	cp $(K8SROOT)/api/openapi-spec/swagger.json gen-apidocs/config/swagger.json
```

Comment:
@tengqm , If the Open API k8s spec (swagger.json) is copied from k8s/k8s, is it necessary to keep a copy of this file
in this repo? Could the file copy to the temporary build location?

Reply:
Ideally, we should avoid caching a copy of the swagger.json file. However, we observed that the upstream swagger.json file may get changed after a release is out. In other words, swagger.json from 1.17.0 may differ from that from 1.17.1. We are tracking a release branch say `release-1.17`, not a specific tag. The implication is that we are shooting at a moving target. Keeping a local copy is a simplification of the process. We effectively only care about versions like 1.16.0, 1.17.0. During the release week, we update the JSON once, and adapt the generator accordingly. After that, during the whole development cycle of the next release, we rarely update the JSON.

Reply:
OK. @tengqm , I would like this sorted out so that the spec is automatically updated as part of the release (see my changes to the Makefile, see https://github.com/kubernetes/website/pull/18067.). 
Personally, I think it is important to support a simplified way of generating the references -- one k8s/website "wrapper" build/setup script, one time setup of repos to build, and minimal or no manual script updates for versioning.
I plan on updating the Makefile once the restructuring is completed and will need to fix the command that copies the spec from k8s/k8s (add checks before copying).

Two ideas to address versioning of `swagger.json`.
Copy the release version of the file to this repo from k8s/k8s and:
- copy/rename the file to a versioned filename, for example, (config_dir/swagger.json_1.17). As part of reference generation, this file is copied to <some location>/`swagger.json`. The script/Makefile checks if the versioned file in this repo exists before copying from k8s/k8s.
- identify as a file in a versioned directory.

In the case that there is an error in the API (not a generation error) and the API reference needs to be regenerated, a new, different versioned Open API spec will be used and replace the previously copied version?

Reply:
Sounds a good idea to me. Adding to your proposal, maybe we should couple the config.yaml file with the corresponding swagger JSON file. The reason, as you may have noticed, is that resources are added and removed (deprecated) from one version to another version. A versioned directory for both files sounds a solution to me.

Reply:
👍